PermanentOrg/web-app
# Upgrade to Angular 12

This PR upgrades our Angular version from Angular 11 to Angular 12. This includes upgrading Angular's core libraries to version 12, as well as upgrading peer dependencies to versions that are compatible to Angular 12 (such as upgrading `shallow-render` to a newer version that supports Angular 12). Angular-CLI automates some of the migration process, and these automated changes are included in their own commit. In addition, some manual changes needed to be made to get everything working.

To test:
- Make sure to run `npm install`
- Run tests, the local dev proxy, and a production build
- Make sure the app is working

At the current time, one issue is that we do get a bunch of deprecation warnings from Sass when running any builds/tests/dev servers. However, these are currently not breaking changes, only deprecation warnings. Do we want to tackle that in this PR or just add it as an issue to be resolved separately from the main upgrade?
Note: A lot of these deprecation warnings are related to #127 and would be resolved by fixing it.

Resolves PER-8853: Upgrade to Angular 12. This PR also resolves #131, but not the other dependabot PRs at this time. Also resolves #137.
